src/daemon/daemon.cpp

```diff
@@ -326,29 +333,51 @@ QString Daemon::logs() {
 
 void Daemon::cleanLogs() { LogHandler::instance()->cleanupLogs(); }
 
-bool Daemon::switchServer(const InterfaceConfig& config) {
-  // Generic server switching is not supported without wgutils.
-  if (!supportWGUtils()) {
-    qFatal("Have you forgotten to implement switchServer?");
+bool Daemon::supportServerSwitching(const InterfaceConfig& config) const {
+  if (!m_connections.contains(config.m_hopindex)) {
     return false;
   }
+  const InterfaceConfig& current =
+      m_connections.value(config.m_hopindex).m_config;
+
+  return current.m_privateKey == config.m_privateKey &&
```

Comment:
we could implement an `operator==()` or a similar method.

Reply:
This is not quite the same as testing for equality, it's really testing that the bits that would go into the `[Interface]` section of a wg-quick config file are unchanged but allows for the peers to be different.